src/newButton/Button.tsx

```diff
@@ -0,0 +1,234 @@
+import {
+  Button as ReakitButton,
+  ButtonProps as ReakitButtonProps,
+} from "reakit";
+import * as React from "react";
+import { cx } from "@renderlesskit/react";
+
+import { runIfFn } from "../utils";
+import { useTheme } from "../theme";
+import { Spinner } from "../spinner";
+import { Dict, forwardRefWithAs } from "../utils/types";
+
+export type ButtonProps = Omit<ReakitButtonProps, "prefix"> & {
+  /**
+   * How large should the button be?
+   *
+   * @default md
+   */
+  size?: keyof Renderlesskit.GetThemeValue<"newButton", "size">;
+  /**
+   * How the button should look?
+   *
+   * @default solid
+   */
+  variant?: keyof Renderlesskit.GetThemeValue<"newButton", "variant">;
+  /**
+   * If added, the button will only show an icon ignoring other childrens.
+   */
+  iconOnly?: React.ReactElement;
+  /**
+   * If added, the button will show an icon before the button's text.
+   */
+  suffix?: React.ReactElement;
+  /**
+   * If added, the button will show an icon before the button's text.
+   */
+  prefix?: React.ReactElement;
+  /**
+   * If `true`, the button will be disabled.
+   *
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the button will show a spinner.
+   *
+   * @default false
+   */
+  loading?: boolean;
+  /**
+   * If added, the button will show this spinner components
+   */
+  spinner?: React.ReactElement;
+};
+
+export const withIconA11y = (icon: React.ReactElement, props?: Dict) => {
```

Comment:
Lets extract this to utils file.

Reply:
Right, I will extract them out to `utils`